src/ocean/Compute.ts

```diff
@@ -244,39 +243,60 @@ export class Compute extends Instantiable {
       did,
       consumerAccount,
       jobId,
-      txId,
-      sign
+      txId
     )
 
     return computeJobsList as ComputeJob[]
   }
 
   /**
-   * Returns the final result of a specific compute job published as an asset.
+   * Downloads compute job result
    * @param  {Account} consumerAccount The account of the consumer ordering the service.
-   * @param  {DDO|string} asset DID Descriptor Object containing all the data related to an asset or a Decentralized identifier.
-   * @param  {string} jobId The ID of the compute job to be stopped.
-   * @return {Promise<ComputeJob>} Returns the DDO of the result asset.
+   * @param  {string} jobId JobId
+   * @param  {number} index Compute Result index
+   * @param  {string} destination: destination folder
+   * @param  {string} did Decentralized identifier.
+   * @param  {DDO} ddo If undefined then the ddo will be fetched by did, this is just to optimize network calls
+   * @param  {ServiceCompute} service If undefined then we get the service from the ddo
```

Comment:
where else would we get it from? Why can another compute service be passed to this method which seems to be then different from the DDO compute service? What's the use case for this?

Reply:
it's for use cases where you already have the service, but not the ddo. This was actually from the market

Reply:
If I see this right, we always have full DDOs in this `getComputeJobs` method:
https://github.com/oceanprotocol/market/blob/main/src/utils/compute.ts#L257

we just don't pass it as part of `ComputeJobMetaData` but we actually could:
https://github.com/oceanprotocol/market/blob/main/src/utils/compute.ts#L188

Cause then we could remove that passing only a service use case from here. By always requiring passing of an `asset` which then is either a `DDO` or just `string/DID` we have made 3 parameters into 1 too